services/pdl_service/pdl_service.py

```diff
@@ -0,0 +1,48 @@
+from watchdog.observers import Observer
+from watchdog.events import LoggingEventHandler
+
+from baselayer.log import make_log
+from baselayer.app.models import init_db
+from baselayer.app.env import load_env
+
+from skyportal.handlers.api.earthquake import post_earthquake_from_xml
+from skyportal.models import (
+    DBSession,
+)
+
+env, cfg = load_env()
+
+init_db(**cfg['database'])
+
+log = make_log('pdlserver')
+
+
+def service():
+    while True:
+        try:
+            user_id = 1
+            path = "services/pdlclient_service/data/receiver_storage/origin"
+            with DBSession() as session:
+
+                class Event(LoggingEventHandler):
+                    def on_modified(self, event):
+                        if "quakeml.xml" in event.src_path:
+                            with open(event.src_path) as fid:
+                                payload = fid.read()
+                            post_earthquake_from_xml(payload, user_id, session)
+
+                event_handler = Event()
+                observer = Observer()
+                observer.schedule(event_handler, path, recursive=True)
+                observer.start()  # for starting the observer thread
+                observer.join()
```

Comment:
Shouldn't there be a small pause between iterations on this endless loop?

Reply:
The watchdog seems to take care of this, and the rate of new events is low enough that I don't expect us to be overwhelmed.